runtime/util/cphelp.c

```diff
@@ -239,17 +246,19 @@ addJarToSystemClassLoaderClassPathEntries(J9JavaVM *vm, const char *filename)
 			}
 		}
 #endif
-		/* Everything OK, install the new array and discard the old one */
-		TRIGGER_J9HOOK_VM_CLASS_LOADER_CLASSPATH_ENTRY_ADDED(vm->hookInterface, vm, classLoader, cpEntry);
 		newCount = entryCount + 1;
 		classLoader->classPathEntries = newEntries;
 		classLoader->classPathEntryCount = newCount;
 		j9mem_free_memory(oldEntries);
 	}
 done:
+	omrthread_monitor_exit(classLoader->cpEntriesMutex);
 	/* If any error occurred, discard any allocated memory and throw OutOfMemoryError */
 	if (0 == newCount) {
 		j9mem_free_memory(newEntries);
+	} else {
+		/* Everything OK, install the new array and discard the old one. Safer to put the hook outside of mutex. */
```

Comment:
"install the new array and discard the old one" should stay above where it's done.